app/controllers/sessions_controller.rb

```diff
@@ -30,25 +30,44 @@ def create
     redirect_to url
   end
 
+  def lti_setup
+    consumer_key  = request.params["oauth_consumer_key"]
+    shared_secret = LtiConfiguration
+      .find_by(consumer_key: consumer_key)
+      .shared_secret
+
+    strategy = request.env["omniauth.strategy"]
+    strategy.options.consumer_key = consumer_key
+    strategy.options.shared_secret = shared_secret
+
+    head :ok
+  end
+
+  # rubocop:disable MethodLength
+  # rubocop:disable AbcSize
   def lti_launch
```

Comment:
It looks like there's some untested logic in this method, would you mind adding some tests to fix that up?